src/lib.rs

```diff
@@ -229,16 +226,46 @@ impl Navigator {
         self.bmp.pressure_kpa().unwrap()
     }
 
-    pub fn read_adc(&mut self) -> ADCData {
+    pub fn read_adc_all(&mut self) -> ADCData {
         ADCData {
             channel: [
-                block!(self.adc.read(&mut channel::SingleA0)).unwrap(),
-                block!(self.adc.read(&mut channel::SingleA1)).unwrap(),
-                block!(self.adc.read(&mut channel::SingleA2)).unwrap(),
-                block!(self.adc.read(&mut channel::SingleA3)).unwrap(),
+                block!(self.adc.read(&mut ads1x1x::channel::SingleA0)).unwrap(),
+                block!(self.adc.read(&mut ads1x1x::channel::SingleA1)).unwrap(),
+                block!(self.adc.read(&mut ads1x1x::channel::SingleA2)).unwrap(),
+                block!(self.adc.read(&mut ads1x1x::channel::SingleA3)).unwrap(),
             ],
         }
     }
+
+    pub fn read_adc(&mut self, channel: adc_Channel) -> i16 {
+        match channel {
+            adc_Channel::SingleA0 => {
+                block!(self.adc.read(&mut ads1x1x::channel::SingleA0)).unwrap()
+            }
+            adc_Channel::SingleA1 => {
+                block!(self.adc.read(&mut ads1x1x::channel::SingleA1)).unwrap()
+            }
+            adc_Channel::SingleA2 => {
+                block!(self.adc.read(&mut ads1x1x::channel::SingleA2)).unwrap()
+            }
+            adc_Channel::SingleA3 => {
+                block!(self.adc.read(&mut ads1x1x::channel::SingleA3)).unwrap()
+            }
+            adc_Channel::DifferentialA0A1 => {
+                block!(self.adc.read(&mut ads1x1x::channel::DifferentialA0A1)).unwrap()
+            }
+            adc_Channel::DifferentialA0A3 => {
+                block!(self.adc.read(&mut ads1x1x::channel::DifferentialA0A3)).unwrap()
+            }
+            adc_Channel::DifferentialA1A3 => {
+                block!(self.adc.read(&mut ads1x1x::channel::DifferentialA1A3)).unwrap()
+            }
+            adc_Channel::DifferentialA2A3 => {
+                block!(self.adc.read(&mut ads1x1x::channel::DifferentialA2A3)).unwrap()
+            }
+        }
+    }
```

Comment:
why not just making it:
```rs
    pub fn read_adc(&mut self, channel: adc_Channel) -> i16 {
        block!(self.adc.read(&mut channel.into())).unwrap()
    }
```

Reply:
@patrickelectric true, and it seems to be possible to just use the enum itself, as shown [here](https://github.com/eldruin/ads1x1x-rs/blob/master/examples/trait.rs#L12):

```Rust
    pub fn read_adc(&mut self, channel: adc_Channel) -> i16 {
        block!(self.adc.read(channel)).unwrap()
    }
```

Reply:
Good, changed the imports, using DynamicOneShot the trait cover enum inputs